Fix segfault while real and imaginary attributes are set to a number

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73867

Fixes https://github.com/pytorch/pytorch/issues/72947

Differential Revision: [D34695956](https://our.internmc.facebook.com/intern/diff/D34695956)

anjali411 committed:
Fix segfault while real and imaginary attributes are set to a number

[ghstack-poisoned]

vadimkantorov commented on torch/csrc/autograd/python_variable.cpp:
```diff
@@ -1057,28 +1058,28 @@ PyObject *THPVariable_get_imag(THPVariable* self, void *unused)
   auto& self_ = THPVariable_Unpack(self);
-  auto& real_ = THPVariable_Unpack(real);
+  auto self_real = at::real(self_);
+  auto real_ = valueToTensor(self_real.options(), real, self_real.device());
```
will this materialize the `value` tensor to the shape or `real`?

anjali411 replied:
no it just creates a zero dimensional tensor

anjali411 committed:
Update on "Fix segfault while real and imaginary attributes are set to a number"


Fixes https://github.com/pytorch/pytorch/issues/72947

[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34695956).

anjali411 committed:
Update on "Fix segfault while real and imaginary attributes are set to a number"


Fixes https://github.com/pytorch/pytorch/issues/72947

Differential Revision: [D34695956](https://our.internmc.facebook.com/intern/diff/D34695956)

[ghstack-poisoned]

anjali411 committed:
Update on "Fix segfault while real and imaginary attributes are set to a number"


Fixes https://github.com/pytorch/pytorch/issues/72947

Differential Revision: [D34695956](https://our.internmc.facebook.com/intern/diff/D34695956)

[ghstack-poisoned]

anjali411 commented:
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34695956).
